mmdet/datasets/custom.py

```diff
@@ -178,12 +178,14 @@ def evaluate(self,
             scale_ranges (list[tuple] | None): Scale ranges for evaluating mAP.
                 Default: None.
         """
+        metrics = metric if isinstance(metric, list) else [metric]
```

Comment:
Thanks for the fix! Since "mAP" and "recall" cannot be evaluated at the same time, we may simply cast metric to a str like in [voc dataset](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/datasets/voc.py#L30-L32).
```python
if not isinstance(metric, str):
    assert len(metric) == 1
    metric = metric[0]
```

Reply:
Thank you! I did not immediately realized that recall and mAP accept different parameters. I will make changes and resubmit.